src/DaedalusCompiler/Compilation/AssemblyBuilder.cs

```diff
@@ -278,14 +326,37 @@ public class FuncCallContext
     {
         public List<DatSymbolType> ParametersTypes;
         public int ArgIndex;
-        public readonly FuncCallContext Parent;
+        public FuncCallContext Parent;
 
-        public FuncCallContext(FuncCallContext parent)
+        public FuncCallContext(FuncCallContext parent=null)
         {
             ParametersTypes = new List<DatSymbolType>();
             ArgIndex = -1;
             Parent = parent;
         }
+
+        public static FuncCallContext Clone(FuncCallContext ctx)
```

Comment:
instead of clone, please make new contructor which as a parameter takes FuncCallContext, it's called copy constructor ;)